### PR TITLE
Add Ethereum Tooling Guides

### PR DESCRIPTION
**Pull request by jolube**
Closes: #XXX

## Description

This PR improves on existing Ethereum tooling guides (Truffle, Remix) in addition to adding new ones (Hardhat, Metamask). The guides now also target testnet configs.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: docs/guides/README.md [hunk 1]
```diff
@@ -13,7 +13,9 @@ parent:
   * [Keyring](./keys-wallets/keyring)
   * [Metamask](./keys-wallets/metamask)
```
Comment (fedekunze) [inline docs/guides/README.md:14]: should we remove this one?

File: docs/guides/tools/hardhat.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+$ node -v
+...
+
+$ npm -v
+...
+```
+
+:: tip
+If you haven't already, you will also need to install Ethermint if you plan on deploying your smart contracts locally. Check this [document](./../../quickstart/installation.md) for the full instructions.
+::
```
Comment (fedekunze) [inline docs/guides/tools/hardhat.md:50]:
```suggestion
::: tip
If you haven't already, you will also need to install Ethermint if you plan on deploying your smart contracts locally. Check this [document](./../../quickstart/installation.md) for the full instructions.
:::
```

File: docs/guides/tools/metamask.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+MetaMask should now be connected to your local Ethermint node via its web3 RPC, and you should have a balance of 0 {{ $themeConfig.project.ticker }}.
+
+{IMAGE HERE}
+
+## Import Existing Account into MetaMask
+
+If you would like to use an existing local account, you can export its private key using the following command:
+```shell
+$ ethermintd keys unsafe-export-eth-key mykey --keyring-backend test
+```
```
Comment (fedekunze) [inline docs/guides/tools/metamask.md:58]:
```suggestion

```shell
ethermintd keys unsafe-export-eth-key mykey --keyring-backend test
```
```

File: docs/guides/tools/metamask.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<!--
+order: 1
+-->
+
+# MetaMask: Connecting to Ethermint
```
Comment (fedekunze) [inline docs/guides/tools/metamask.md:5]: how does this guide relate to the one on `guides/keys-wallets/`?

Comment (fedekunze) [reply docs/guides/tools/metamask.md:5]: wondering if we should merge them

File: docs/guides/tools/truffle.md [hunk 5]
```diff
@@ -1,49 +1,51 @@
...
+[Truffle](https://www.trufflesuite.com/truffle) is a development framework for deploying and managing [Solidity](https://github.com/ethereum/solidity) smart contracts.
 
-## Install dependencies
+## Install Dependencies
 
 First, install the latest Truffle version on your machine globally.
 
-```bash
-yarn install truffle -g
+```shell
```
Comment (fedekunze) [inline docs/guides/tools/truffle.md:20]: curious about the use of bash vs shell? should we enforce one over the other?

File: docs/guides/tools/truffle.md [hunk 6]
```diff
@@ -1,49 +1,51 @@
...
+$ yarn install truffle -g
 ```
 
-You will also need to install Ethermint. Check this [document](./../../quickstart/installation.md) for the full instructions.
+:: tip
+If you haven't already, you will also need to install Ethermint if you plan on deploying your smart contracts locally. Check this [document](./../../quickstart/installation.md) for the full instructions.
+::
```
Comment (fedekunze) [inline docs/guides/tools/truffle.md:26]:
```suggestion
::: tip
If you haven't already, you will also need to install Ethermint if you plan on deploying your smart contracts locally. Check this [document](./../../quickstart/installation.md) for the full instructions.
:::
```

**Review by fedekunze — APPROVED** [review]
Looks good! Love the tabs too. Left a few questions too (specially around the 2 metamask guides)

I'd suggest you download the `markdownlint` plug-in (if you are using VS Code): https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

File: docs/guides/tools/metamask.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+::: tab Testnet
+- Network Name: `{{ $themeConfig.project.name }}`
+- RPC URL: `{{ $themeConfig.project.rpc_url }}`
+- ChainID: `{{ $themeConfig.project.chain_id }}`
+- Symbol (Optional): `{{ $themeConfig.project.ticker }}`
+- Block Explorer (Optional): `{{ $themeConfig.project.block_explorer_url }}`
+:::
+::::
+
+{IMAGE HERE}
```
Comment (fedekunze) [inline docs/guides/tools/metamask.md:47]: TODO?

File: docs/guides/tools/metamask.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+MetaMask should now be connected to your local Ethermint node via its web3 RPC, and you should have a balance of 0 {{ $themeConfig.project.ticker }}.
+
+{IMAGE HERE}
```
Comment (fedekunze) [inline docs/guides/tools/metamask.md:51]: ditto

File: docs/guides/tools/metamask.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+If you would like to use an existing local account, you can export its private key using the following command:
+
+```shell
+$ ethermintd keys unsafe-export-eth-key mykey --keyring-backend test
+```
+
+You should be given the option to `Import Account` under MetaMask's `My Accounts` menu. Upon completing the import, you should see your account's balance.
+
+{IMAGE HERE}
```
Comment (fedekunze) [inline docs/guides/tools/metamask.md:63]: ditto

File: docs/guides/tools/truffle.md (hunk 5 above)
Comment (jolube) [reply docs/guides/tools/truffle.md:20]: Actually, after doing some research the answer is no. The lexers Github-flavored md uses are different than vuepress it seems.

`shell` is just an alias of `bash ` for vuepress, so using `bash` is fine. Full list [here](https://github.com/PrismJS/prism/blob/master/components.json).

File: docs/guides/README.md (hunk 1 above)
Comment (jolube) [reply docs/guides/README.md:14]: Kept this one and merged the other into it

File: docs/guides/tools/metamask.md (hunk 4 above)
Comment (jolube) [reply docs/guides/tools/metamask.md:5]: Merged

File: docs/guides/tools/metamask.md (hunk 7 above)
Comment (jolube) [reply docs/guides/tools/metamask.md:47]: Deleted this file, so not relevant anymore

File: docs/guides/tools/metamask.md (hunk 8 above)
Comment (jolube) [reply docs/guides/tools/metamask.md:51]: Deleted this file, so not relevant anymore

File: docs/guides/tools/metamask.md (hunk 9 above)
Comment (jolube) [reply docs/guides/tools/metamask.md:63]: Deleted this file, so not relevant anymore

**Pull request merged**